samcli/commands/local/lib/swagger/parser.py

```diff
@@ -123,11 +128,21 @@ def get_authorizers(self, event_type: str = Route.API) -> Dict[str, Authorizer]:
             )
 
             validation_expression = authorizer_object.get(SwaggerParser._AUTHORIZER_LAMBDA_VALIDATION)
-            if event_type != Route.HTTP:
+            if event_type == Route.HTTP and validation_expression:
                 validation_expression = None
 
                 LOG.warning(
-                    "Validation expressions is only available on Rest APIs, ignoring for Lambda authorizer '%s'",
+                    "Validation expressions is only available on REST APIs, ignoring for Lambda authorizer '%s'",
+                    auth_name,
+                )
```

Comment:
This part confuses me a little bit. We were giving the warning message if `event_type` was not `Route.HTTP` but it is the opposite now. Was this a bug that we are fixing here or are we changing the behavior?

Reply:
Yup, this was a bug that was introduced in an earlier PR I made. Validation expression is a regular expression that users can define to validate their authentication header values. This is only supported on REST APIs. For HTTP APIs, this is not a valid field. The tests have also been updated to test the correct behaviour now.

There is no customer impact since this is on a feature branch and is not actually live.